### PR TITLE
Only show link shares for the current user

### LINKED ISSUE #22275: Link shares are not unique per user in the web UI
1. As user test1 share a folder with user2
2. As user test2 share the folder as link
3. As user test1 share the folder as link

### Expected
It should work

### Actual
Step 3 is only possible on OCS, because the web UI shows the link share of test2 already.

### Comment
I think it is actually nice, because it gives you a web view that can delete the link shares of others, but up to the feature maintainers

cc @rullzer

### PR DESCRIPTION
**Pull request by rullzer**
Currently we have no way to display multiple links in the UI.
So just display the link share for the current user.

Fixes #22275 

CC: @nickvergessen @PVince81 @schiesbn @MorrisJobke

### CODE REVIEW

**Comment by PVince81** [comment]
:+1: makes sense

Note: the web UI was never intended to display more than one link share. Maybe it will change one day

**Comment by rullzer** [comment]
> Note: the web UI was never intended to display more than one link share. Maybe it will change one day

Yeah I know. We need to think about that.

Even mutliple link shares for 1 user would eventually make sense in my view... since i want to share a link to a user... so I can unshare when I want. But that is a whole other discussion ;)

**Comment by MorrisJobke** [comment]
The described behavior is now in place, but it looks weird, because the symbol in the files table shows the link icon even if the user didn't have a link share (but other users do).

**Comment by PVince81** [comment]
The link icon come from the summarized response from ajax.php (loadIcon), is the new API returning link shares there ? Make sure it's filitered to the current user. (for now)

**Comment by rullzer** [comment]
aaah warg... stupid ajax.... Let me see if that is fixable...

**Comment by rullzer** [comment]
Ok fixed. I'm not really happy with this all... but I think for now it is the best we can do...

**Comment by rullzer** [comment]
aaah mmm... this does mean that is a file is unshared from a user we should remove all their link shares...

**Comment by rullzer** [comment]
Will add

**Comment by rullzer** [comment]
Aaah mm no... that won't scale properly... since if user1 shares a file with a group with 1000 users we have to check for each of those users if have to remove the share.

Mmmm let me think about this...

File: lib/private/share/share.php [hunk 1]
```diff
@@ -2495,7 +2495,8 @@ private static function createSelectStatement($format, $fileDependent, $uidOwner
 			if ($fileDependent) {
 				$select = '`*PREFIX*share`.`id`, `*PREFIX*share`.`parent`, `share_type`, `path`, `storage`, '
 					. '`share_with`, `uid_owner` , `file_source`, `stime`, `*PREFIX*share`.`permissions`, '
-					. '`*PREFIX*storages`.`id` AS `storage_id`, `*PREFIX*filecache`.`parent` as `file_parent`';
+					. '`*PREFIX*storages`.`id` AS `storage_id`, `*PREFIX*filecache`.`parent` as `file_parent`, '
+					. 'uid_initiator';
```
Comment (nickvergessen) [inline lib/private/share/share.php]: missing \` around the column name

Comment (rullzer) [inline lib/private/share/share.php]: thanks. fixed...

**Comment by rullzer** [comment]
Issue of the invisible links raised in #22295

Lets get this in.

**Comment by MorrisJobke** [comment]
Now the overview never shows the chain icon and only updates the chain icon (in the folder thumbnail and the inline text) when the sidebar is opened and the shares are loaded.

**Comment by MorrisJobke** [comment]
Before opening the sidebar:
bildschirmfoto 2016-02-10 um 17 23 13

Clicking on the share text (which opens the sidebar and loads the shares):
bildschirmfoto 2016-02-10 um 17 23 22

**Comment by rullzer** [comment]
@MorrisJobke mmm strange all works here...

**Comment by PVince81** [comment]
Followed the steps from https://github.com/owncloud/core/issues/22275, works so far.

But regarding the indicator: refresh the page as user1, the indicator shows a link share, which is correct.
However do the same for user2, the indicator shows regular share, not link share.

@rullzer maybe you mixed up users when testing ?

**Comment by PVince81** [comment]
Actually the indicator isn't even for outgoing share, it's from the incoming share.

The getItemStatuses call returns an empty set.

**Comment by PVince81** [comment]
Regarding the original issue, I also tested resharing a subdir: user1 shared "test" with user2, user2 shares "test/sub" with link, then user1 shares "test/sub" with link too. It works correctly, every user sees the correct link share.

**Comment by rullzer** [comment]
Yes I see this now as well. It is due to the new reshare behaviour. If you check on master there is also no link icon..

**Comment by rullzer** [comment]
Ok lets move forward with this PR. The issue Morris found was actually a differnt issue that is also present on current master. It is a result of using the old ajax endpoint for the share info. I'll raise a separate ticket for this.

This does fix the issue it was ment to fix.

**Comment by PVince81** [comment]
I agree with @rullzer, the issue of seeing the wrong link share is much more critical than the missing indicator.

:+1: 

@MorrisJobke do you agree ?

**Comment by LukasReschke** [comment]
Seems to work. Code looks as good as code in `share.php` can look :wink: :+1:

**Pull request merged**